DM-1272 Updated Software section in Device Management Library chapter in Reference guide after changes to Advanced Software Management (#661)
SoftwareAG/c8y-docs: content/reference/device-management-library-bundle/software.md

### Installed software

The installed software packages are listed in `c8y_SoftwareList` fragment which may be placed in the device managed
object or in the single child addition of type `c8y_InstalledSoftwareList`. Let's call the first approach *legacy*

> and the second *advanced*.

Author: I named it "legacy", but I don't thing there is strong encouraging to use "advanced" approach.

### Advanced Software Management

> Using the "advanced" approach, the `c8y_SoftwareList` fragment is placed in the single child addition of type `c8y_InstalledSoftwareList`.
> To facilitate the management of listed software packages, the Advanced Software Management default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices) was introduced.

Reviewer: We should not be pointing users to the data model on the inventory API. I'd describe it like this:
```suggestion
Using the "advanced" approach, the `c8y_SoftwareList` fragment is is no longer present in the device managed object. The data is separated from the device managed object which keeps the size of the device managed object low even for very large lists of installed software. All installed software for a device can be read and managed through the Advanced Software Management default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices).
```

Author: Isn't the whole chapter pointing users to data model on the inventory API?

Reviewer: The point of this documentation is to describe how device integrators can manage their installed software with the platform. In advanced software management this should always happen through the ASM microservice endpoints and never through the standard platform APIs. If there are other places in this chapter that point to inventory API the those need to be changed too.

### Advanced Software Management

> In this approach the `c8y_SoftwareList` fragment is placed in the single child addition of type`c8y_InstalledSoftwareList`.

Reviewer 2: ```suggestion
Using the "advanced" approach, the `c8y_SoftwareList` fragment is placed in the single child addition of type `c8y_InstalledSoftwareList`.
```

> To facilitate management of listed software packages, the Advanced Software Management default
> [microservice](https://cumulocity.com/guides/concepts/applications/#microservices) was introduced.

Reviewer 2: ```suggestion
To facilitate the management of listed software packages, the Advanced Software Management default [microservice](https://cumulocity.com/guides/concepts/applications/#microservices) was introduced.
```

> Devices may indicate their support for Advanced Software Management by including the ```c8y_SoftwareUpdate``` operation

Reviewer 2: ```suggestion
Devices support Advanced Software Management when they include the `c8y_SoftwareUpdate` operation
```

> in their ```c8y_SupportedOperations``` fragment and listing their supported software types in
> the ```c8y_SupportedSoftwareTypes``` fragment.

Reviewer 2: ```suggestion
in their `c8y_SupportedOperations` fragment and list their supported software types in the `c8y_SupportedSoftwareTypes` fragment.
```
